### PR TITLE
fix: Remove dead code detected by the rust beta compiler

### PR DESCRIPTION
**Pull request by flub**
## Description

The beta compiler identifies these as dead code.

## Change checklist

- [x] Self-review.

### CODE REVIEW

File: iroh-sync/src/ranger.rs [hunk 1]
```diff
@@ -40,13 +40,13 @@ pub trait RangeEntry: Debug + Clone {
...
+    // fn is_prefix_of(&self, other: &Self) -> bool;
 
-    /// Returns true if `other` is a prefix of `self`.
-    fn is_prefixed_by(&self, other: &Self) -> bool {
-        other.is_prefix_of(self)
-    }
+    // /// Returns true if `other` is a prefix of `self`.
+    // fn is_prefixed_by(&self, other: &Self) -> bool {
+    //     other.is_prefix_of(self)
+    // }
```
Comment (flub) [inline iroh-sync/src/ranger.rs:49]: @Frando What is the deal with this trait?

Comment (flub) [reply iroh-sync/src/ranger.rs:49]: oh, it's used in tests

Comment (flub) [reply iroh-sync/src/ranger.rs]: @Frando I think I've made this happy for the compiler.  But could you make sure it does the right thing?

**Comment by flub** [comment]
@Frando  could you, err,  approve in that case?

File: iroh-bytes/src/store/traits.rs [hunk 2]
```diff
@@ -216,43 +216,6 @@ impl<W: BaoBatchWriter, F: Fn(u64, usize) -> io::Result<()> + 'static> BaoBatchW
     }
 }
 
-/// A combined batch writer
```
Comment (rklaehn) [inline iroh-bytes/src/store/traits.rs:219]: Yeah, this is dead since we have a proper batch writer. In some branch I removed this already.

**Review by rklaehn — APPROVED** [review]

**Review by Frando — APPROVED** [review]

**Pull request merged**